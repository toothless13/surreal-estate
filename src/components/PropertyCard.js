import React from "react";

const PropertyCard = ({ title, type, bedrooms, bathrooms, price }) => {
  return (
    <div className="property-card">
      <p className="property-card-title">{title}</p>
      <p className="property-card-type">{type}</p>
      <p className="property-card-bedrooms" data-testid="property-bedrooms">
        {bedrooms}
      </p>
      <p className="property-card-bathrooms" data-testid="property-bathrooms">
        {bathrooms}
      </p>
      <p className="property-card-price">{price}</p>
    </div>
  );
};

export default PropertyCard;
