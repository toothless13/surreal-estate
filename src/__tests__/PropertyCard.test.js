import React from "react";
import { render, screen } from "@testing-library/react";
// import { toBeInTheDocument } from "@testing-library/jest-dom/matchers";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "Test Title",
    type: "Test Type",
    bedrooms: 3,
    bathrooms: 1,
    price: 12345,
    city: "Test City",
    email: "test@email.com",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bedrooms={validProps.bedrooms}
        bathrooms={validProps.bathrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct value for title prop", () => {
    const component = render(<PropertyCard title={validProps.title} />);

    expect(component.getByText(/Test Title/i)).toBeTruthy();
    expect(component.getByText(/Test Title/i)).toBeInTheDocument();
  });

  it("renders correct value for type prop", () => {
    const component = render(<PropertyCard type={validProps.type} />);

    expect(component.getByText(/Test Type/i)).toBeTruthy();
    expect(component.getByText(/Test Type/i)).toBeInTheDocument();
  });

  it("renders correct value for bedrooms prop", () => {
    const { getByTestId } = render(
      <PropertyCard bedrooms={validProps.bedrooms} />,
    );

    expect(getByTestId("property-bedrooms")).toBeTruthy();
    expect(getByTestId("property-bathrooms")).toBeInTheDocument();
  });

  it("renders correct value for bathrooms prop", () => {
    const { getByTestId } = render(
      <PropertyCard bathrooms={validProps.bathrooms} />,
    );

    expect(getByTestId("property-bathrooms")).toBeTruthy();
  });

  it("renders correct value for price prop", () => {
    const component = render(<PropertyCard price={validProps.price} />);

    expect(component.getByText(/12345/i)).toBeTruthy();
    expect(component.getByText(/12345/i)).toBeInTheDocument();
  });
});
