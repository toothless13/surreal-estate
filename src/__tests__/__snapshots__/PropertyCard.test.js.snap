// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <p
      class="property-card-title"
    >
      Test Title
    </p>
    <p
      class="property-card-type"
    >
      Test Type
    </p>
    <p
      class="property-card-bedrooms"
      data-testid="property-bedrooms"
    >
      3
    </p>
    <p
      class="property-card-bathrooms"
      data-testid="property-bathrooms"
    >
      1
    </p>
    <p
      class="property-card-price"
    >
      12345
    </p>
  </div>
</DocumentFragment>
`;
